### PR TITLE
Get-DbaBackupHistory, allow filtering by type

### PR DESCRIPTION
**Pull request by niphlod**
Changes proposed in this pull request:
 - Introduced Type filter to speedup full history retrieval

### CODE REVIEW

**Comment by potatoqualitee** [comment]
This is stuart's baby - I'mma let him review

File: functions/Get-DbaBackupHistory.ps1 [hunk 1]
```diff
@@ -237,15 +257,21 @@ function Get-DbaBackupHistory {
...
+				foreach ($db in $databases) {
 					
 					#Get the full and build upwards
 					$allbackups = @()
-					$allbackups += $Fulldb = Get-DbaBackupHistory -SqlInstance $server -Database $db -LastFull -raw:$Raw -DeviceType $DeviceType
-					$DiffDB = Get-DbaBackupHistory -SqlInstance $server -Database $db -LastDiff -raw:$Raw -DeviceType $DeviceType
+					$allbackups += $Fulldb = Get-DbaBackupHistory -SqlInstance $server -Database $db.Name -LastFull -raw:$Raw -DeviceType $DeviceType
+					$DiffDB = Get-DbaBackupHistory -SqlInstance $server -Database $db.Name -LastDiff -raw:$Raw -DeviceType $DeviceType
 					if ($DiffDb.LastLsn -gt $Fulldb.LastLsn -and  $DiffDb.DatabaseBackupLSN -eq $Fulldb.CheckPointLSN ) {
+						$Allbackups += $DiffDB = Get-DbaBackupHistory -SqlInstance $server -Database $db.Name -LastDiff -raw:$Raw -DeviceType $DeviceType
```
Comment (Stuart-Moore) [inline functions/Get-DbaBackupHistory.ps1]: Isn't this repeating the call on L271? Just $Allbackups += $DiffDB could be used?

File: functions/Get-DbaBackupHistory.ps1 [hunk 2]
```diff
@@ -258,18 +284,17 @@ function Get-DbaBackupHistory {
 							continue
 						}
 					}
-					$Allbackups += $Logdb = Get-DbaBackupHistory -SqlInstance $server -Databases $db -raw:$raw -DeviceType $DeviceType -LastLsn $TLogstartLSN | Where-Object { 
+					$Allbackups += $Logdb = Get-DbaBackupHistory -SqlInstance $server -Database $db.Name -raw:$raw -DeviceType $DeviceType -LastLsn $TLogstartLSN | Where-Object { 
 						$_.Type -eq 'Log' -and [bigint]$_.LastLsn -gt [bigint]$TLogstartLSN -and [bigint]$_.DatabaseBackupLSN -eq [bigint]$Fulldb.CheckPointLSN 
-					} | Sort-Object LastLsn
-					$Allbackups
+					}
+					$Allbackups | Sort-Object FirstLsn
```
Comment (Stuart-Moore) [inline functions/Get-DbaBackupHistory.ps1]: Any reason for switching from LastLsn to FirstLsn? (shouldn't make a difference)

**Comment by niphlod** [comment]
Will check and reply in a couple of hours, stay tuned

Il 12 lug 2017 1:06 PM, "Stuart Moore" <notifications@github.com> ha
scritto:

File: functions/Get-DbaBackupHistory.ps1 (hunk 2 above)
Comment (niphlod) [inline functions/Get-DbaBackupHistory.ps1]: seemingly noone, it may have slipped when trying to fix this on garbo. will revert.

**Comment by niphlod** [comment]
done

**Comment by niphlod** [comment]
aaaand example added.

**Pull request merged**